examples/simple.rs

```diff
@@ -1,67 +1,120 @@
-use hyper::server::conn::AddrStream;
-use hyper::service::{make_service_fn, service_fn};
-use hyper::{Body, Request, Response, Server, StatusCode};
+use std::convert::Infallible;
+use std::io;
+use std::net::{IpAddr, SocketAddr};
+use std::time::Duration;
+
+use http_body_util::combinators::UnsyncBoxBody;
+use http_body_util::{BodyExt, Empty, Full};
+use hyper::body::{Bytes, Incoming};
+use hyper::server::conn::http1;
+use hyper::service::service_fn;
+use hyper::{Request, Response, StatusCode};
+use hyper_util::rt::{TokioExecutor, TokioIo, TokioTimer};
+use tokio::net::TcpListener;
+
 use hyper_reverse_proxy::ReverseProxy;
-use hyper_trust_dns::{RustlsHttpsConnector, TrustDnsResolver};
-use std::net::IpAddr;
-use std::{convert::Infallible, net::SocketAddr};
+use hyper_rustls::{ConfigBuilderExt, HttpsConnector};
+use hyper_util::client::legacy::connect::HttpConnector;
+
+type Connector = HttpsConnector<HttpConnector>;
+type ResponseBody = UnsyncBoxBody<Bytes, std::io::Error>;
 
 lazy_static::lazy_static! {
```

Comment:
As long as you're here fixing up the example, I think it's no longer recommended to use lazy_static. From the [lazy_static repo](https://github.com/rust-lang-nursery/lazy-static.rs):

> It is now possible to easily replicate this crate's functionality in Rust's standard library with [std::sync::OnceLock](https://doc.rust-lang.org/std/sync/struct.OnceLock.html).

So if anything it'd be better to use that.

Reply:
Good idea, I opened https://github.com/stefansundin/hyper-reverse-proxy/pull/2.